test/test_shell_interface.rb

```diff
@@ -151,6 +151,74 @@ module Syskit
             end
         end
 
+        describe "the log configuration management" do
+            before { Syskit.conf.logs.create_group 'test' }
+            after { Syskit.conf.logs.remove_group('test') }
+
+            it 'creates a marshallable instance of the configuration' do
+                conf = subject.logging_conf
+                assert (conf.port_logs_enabled == Syskit.conf.logs.port_logs_enabled?)
+                assert (conf.conf_logs_enabled == Syskit.conf.logs.conf_logs_enabled?)
+                Syskit.conf.logs.groups.each_pair do |key, group|
+                    assert (group.enabled? == conf.groups[key].enabled)
+                end
+                Marshal.dump(conf)
+            end
+
+            it "makes a deep copy of the configuration" do
+                one = subject.logging_conf
+                two = one.deep_copy
+
+                assert (one == two)
+                assert !one.equal?(two)
+            end
+
+            it 'triggers a redeployment when the configuration is updated' do
+                conf = subject.logging_conf
+                flexmock(subject).should_receive(:redeploy).once
+                subject.update_logging_conf(conf)
```

Comment:
This tests that redeploy is called, not that the configuration got updated first.